### ISSUE TITLE
Messenger docs refer to an outdated interface

### ISSUE THREAD

**Issue by yivi** [symfony/symfony-docs#11789]
Messenger documentation refers to the `Symfony\Component\Messenger\Transport\Receiver\ReceiverInterface` that was included up to 4.2, but it appears that this interface has greatly changed in 4.3.

https://symfony.com/doc/current/components/messenger.html#your-own-receiver
https://symfony.com/doc/current/messenger/custom-transport.html#create-your-transport-factory

In 4.2 used to be:

```php
interface ReceiverInterface
{
    public function receive(callable $handler): void;
    public function stop(): void;
}
```

Now it is:

```php
interface ReceiverInterface
{
    public function get(): iterable;
    public function ack(Envelope $envelope): void;
    public function reject(Envelope $envelope): void;
}
```

Nothing in common but the name. :(

The custom transport I had in place is no longer compatible with 4.3 (experimental module, I know), but now have to look at the included Transports to figure out how to reimplement this one.

Fleshing out this part of the docs would be great.

**Comment by Guikingone** [comment]
Hi @yivi 👋 

Ok, I've looked at the problem and effectively, the interface has changed since 4.3, for me, here's the actual behaviour: 

`ReceiverInterface::get()` act like `ReceiverInterface::receive(callable $handler)` so if you need to keep the same logic, chances are you're gonna need to reimplement this method. 

The `get()` method is tied to `ack` and `reject` which are similar to other libraries/languages approach when it comes to message handling.

I'm gonna open a PR on the doc to improve the 4.3 version and inform that this interface has changed (I think also that it could be a good idea to trigger a deprecation on 4.2, what do you think?), thanks for the feedback on this BC break (which can be normal as Messenger is experimental) 🙂

**Comment by yivi** [comment]
Thanks for the feedback. My transport has been already reimplemented. I actually prefer the new interface, neater and easier to follow.

Just wanted to point this since the docs still referred to the old interface, which may confuse a user or two. 

Another thing that may worth doing is pointing out somewhere in the docs the component is experimental. I had realized that because I checked the source code, but some users of the module may not realize they are using something that could have this kind of change between versions.

Thanks again. Regards. 👍
